### DISCUSSION TITLE
STM32G030F6P6 I2C problem

### DISCUSSION THREAD

**Discussion by Chicco7** [stm32duino/Arduino_Core_STM32#1696] [Q&A] [answered]
Hi, I have made a custom board with an STM32G030F6P6, and I can't use the hardware I2C. I know that in this particolar chip I can't use the Serial and the I2C1 in the same time because the PB7 pin it's used by both, so I don't use the serial.
I use a u8g2 lib with oled display, it work only if I use the SW definition, but I know that is a software emulation of I2C and don't use the native hardware I2C.  Also I have to use an MCP4725 (DAC Module) in the same I2C and I can't use the library if the HW I2C don't work.
I tried the I2C Scan sketch that use the wire library but don't show anything. It almost looks as if the pins are not mapped.
Anyone have an idea or the same problem?

Thanks

**Comment by fpistm** [comment]
Heve you wired pullup resistors on each I2C lines?

**Reply by Chicco7** [reply]
Yes, a 4,7k resistor each line

**Comment by Chicco7** [comment]
Ok, at least after hours of test I reach the achievement!
I check with I2C scan software and it found the 2 device attached.
I switch to the library Wire but nothing, no device found.
In my board I have Jumper to change the SDA SCL pin to the board because according to the datasheet more then one pin do the same function (in theory).
So I tried to manual set SDA and SCL in Wire library and after various attempts I found that with SDA PB9 and SCL PB8 the 2 device are listed.
At least I tried to set Wire SDA and SCL manually and use the library that I need and all seem work.
I think there is a problem with the Wire library that not initialize the correct pin, and if you don't set manually SDA and SCL it don't work.
I hope this will help other if they need it

**Reply by fpistm** [reply, marked as answer]
I guess you used the `Generic G030F6Px` variant.
The default the Wire instance is:
https://github.com/stm32duino/Arduino_Core_STM32/blob/4e220184c9734313771e69e5646942085fc693a2/variants/STM32G0xx/G030F6P/variant_generic.h#L94-L100

The `_R` is for remap pins. So this is normal it does not work by default and this is the normal behavior. 
As you want use another pins by default, you can:
- create your own variant with the desired mapping
- redefine it a build time using [build_opt.h](https://github.com/stm32duino/wiki/wiki/Customize-build-options-using-build_opt.h) with:
`-DPIN_WIRE_SDA=PB9 -DPIN_WIRE_SCL=PB8`
- change default pins before call the `Wire.Begin()` (see https://github.com/stm32duino/wiki/wiki/API#change-default-wire-instance-pins):
```C++
    Wire.setSDA(PB9);
    Wire.setSCL(PB8);
    Wire.begin();
```
